# <div style="padding:20px;color:white;margin:0;font-size:35px;font-family:Georgia;text-align:left;display:fill;border-radius:5px;background-color:#254E58;overflow:hidden"><b> Day 12- Python Date-time</b></div>

## Python Dates

**A date in Python is not a data type of its own, but we can import a module named datetime to work with dates as date objects.**

### ExampleGet your own Python Server
**Import the datetime module and display the current date:**

In [1]:
import datetime

x = datetime.datetime.now()
print(x)

2023-10-25 19:24:32.814509


## Date Output

* When we execute the code from the example above the result will be:

* 2023-10-25 19:24:32.814509
* The date contains year, month, day, hour, minute, second, and microsecond.

* The datetime module has many methods to return information about the date object.

### Example
### Return the year and name of weekday:**

In [3]:
import datetime

x = datetime.datetime.now()

print(x.year)
print(x.strftime("%A"))

2023
Wednesday


## Creating Date Objects

**To create a date, we can use the datetime() class (constructor) of the datetime module.**

**The datetime() class requires three parameters to create a date: year, month, day.**

### Example

### Create a date object:

In [5]:
import datetime

x = datetime.datetime(2023, 10, 17)

print(x)

2023-10-17 00:00:00


**The datetime() class also takes parameters for time and timezone (hour, minute, second, microsecond, tzone), but they are optional, and has a default value of 0, (None for timezone).**

## The strftime() Method

**The datetime object has a method for formatting date objects into readable strings**

**The method is called strftime(), and takes one parameter, format, to specify the format of the returned string:**

### Example
**Display the name of the month:**

In [7]:
import datetime

x = datetime.datetime(2023, 10, 28)

print(x.strftime("%B"))

October
